pyccel/ast/builtins.py

```diff
@@ -80,7 +80,9 @@ class PythonReal(PythonComplexProperty):
     arg : Variable, Literal
     """
     def __new__(cls, arg):
-        if arg.dtype is not NativeComplex():
+        if isinstance(arg.dtype, NativeBool):
+            return PythonInt(arg)
+        elif arg.dtype is not NativeComplex():
```

Comment:
Is there a reason why we do not write
```python
if arg.dtype is not NativeBool():
```
or
```python
if not isinstance(arg.dtype, NativeComplex):
```
?

Reply:
no, we can work with one of them, its just because i always work with `isinstance`.

Reply:
we should use `isinstance` if the types are no longer singletons, if they are we can use `is`

Reply:
`DataType` objects are singletons

Reply:
@mostafaMamoni Whatever solution you choose, please use the same everywhere in the `if/elif/else` block.